src/topology/builder.rs

```diff
@@ -130,8 +132,12 @@ pub async fn build_pieces(
             }
             Transform::Task(t) => {
                 let filtered = filter_event_type(input_rx, input_type);
+                let decorated = Box::new(filtered.map(move |event| {
```

Comment:
Could be nicer to use `inspect` here rather than `map`.

Reply:
Heck, yes!